core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/helm/HelmChartProcessor.java

```diff
@@ -0,0 +1,224 @@
+package io.quarkiverse.operatorsdk.deployment.helm;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+
+import io.dekorate.helm.model.Chart;
+import io.dekorate.utils.Serialization;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
+import io.quarkiverse.operatorsdk.deployment.AddClusterRolesDecorator;
+import io.quarkiverse.operatorsdk.deployment.ControllerConfigurationsBuildItem;
+import io.quarkiverse.operatorsdk.deployment.GeneratedCRDInfoBuildItem;
+import io.quarkiverse.operatorsdk.runtime.BuildTimeOperatorConfiguration;
+import io.quarkiverse.operatorsdk.runtime.QuarkusControllerConfiguration;
+import io.quarkus.container.spi.ContainerImageInfoBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
+import io.quarkus.deployment.pkg.builditem.ArtifactResultBuildItem;
+import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+import io.quarkus.kubernetes.spi.ConfiguratorBuildItem;
+import io.quarkus.kubernetes.spi.GeneratedKubernetesResourceBuildItem;
+import io.quarkus.qute.Qute;
+
+public class HelmChartProcessor {
+
+    private static final Logger log = Logger.getLogger(HelmChartProcessor.class);
+
+    private static final String TEMPLATES_DIR = "templates";
+    private static final String[] TEMPLATE_FILES = new String[] {
+            "generic-crd-cluster-role.yaml",
+            "generic-crd-cluster-role-binding.yaml",
+            "service.yaml",
+            "serviceaccount.yaml"
+    };
+    public static final String CHART_YAML_FILENAME = "Chart.yaml";
+    public static final String VALUES_YAML_FILENAME = "values.yaml";
+    public static final String CRD_DIR = "crds";
+
+    @BuildStep
+    public void handleHelmCharts(
+            BuildProducer<ArtifactResultBuildItem> dummy,
```

Comment:
If this is needed to ensure this steps runs after another one, please add a comment to that effect, otherwise that parameter might get removed at some point, leading to possibly difficult to diagnose issues.

Reply:
Actually, this is a BuildProducer that isn't used so I don't think there's any point in keeping this.

Reply:
without it it does not gets executed

Reply:
maybe there is a more elegant way to do this